From 1992a82c34c4555b7afb49b4c42eb799d6be9222 Mon Sep 17 00:00:00 2001
From: Per Persson <per.persson@intel.com>
Date: Fri, 6 Mar 2015 10:38:58 +0100
Subject: [PATCH 12/28] atomisp2: Fix hmm memtrack report

Memtrack report is updated to correctly report active
and free memory.

Change-Id: Iad0c95e2e8186cb0da870f29926b5e67e1b1d28b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9279
Original-Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-6624
Original-Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7550
Signed-off-by: Per Persson <per.persson@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/83120
Build: Dessert M COHO (SC)
Build: Dessert M BXT RVP (SC)
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/440498
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm.c        | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm.c
index f61dd6c..358d340 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm.c
@@ -78,8 +78,8 @@ static ssize_t bo_show(struct device *dev, struct device_attribute *attr,
 
 	spin_lock_irqsave(&bo_device.list_lock, flags);
 	list_for_each_entry(bo, bo_list, list) {
-		if ((active && !(bo->status & HMM_BO_FREE)) ||
-			(!active && (bo->status & HMM_BO_FREE))) {
+		if ((active && (bo->status & HMM_BO_ALLOCED)) ||
+			(!active && !(bo->status & HMM_BO_ALLOCED))) {
 			ret = scnprintf(buf + index1, PAGE_SIZE - index1,
 				"%s %d\n",
 				hmm_bo_type_strings[bo->type], bo->pgnr);
-- 
1.9.1

